crates/spk-solve/src/io.rs

```diff
@@ -193,17 +195,167 @@ where
         Ok(())
     }
 
-    fn wait_for_user_to_hit_enter(&self) -> Result<()> {
-        let mut _input = String::new();
+    fn wait_for_user_selection(&self) -> Result<String> {
+        let mut input = String::new();
 
         use std::io::Write;
         let _ = std::io::stdout().flush();
 
-        if let Err(err) = std::io::stdin().read_line(&mut _input) {
+        if let Err(err) = std::io::stdin().read_line(&mut input) {
             // If there's some stdin can't be read, it is probably
             // better to continue with the solve than error out.
             tracing::warn!("{err}");
         }
+        Ok(input)
+    }
+
+    fn show_resolved_packages(&self, state: &Arc<State>) {
+        match state.as_solution() {
+            Err(err) => {
+                tracing::error!("{err}")
+            }
+            Ok(solution) => {
+                // Can't use this without access to the solver repos
+                // and async-ing this method. Could look at enabling
+                // this in future.
+                // solution
+                //     .format_solution_with_highest_versions(
+                //         self.settings.verbosity,
+                //         runtime.solver.repositories(),
+                //         false,
+                //     )
+                //     .await?
+                tracing::info!(
+                    "{}",
+                    format!(
+                        "{}{}",
+                        self.settings.heading_prefix,
+                        solution.format_solution(self.settings.verbosity)
+                    )
+                );
```

Comment:
```suggestion
                tracing::info!(
                  "{}{}",
                  self.settings.heading_prefix,
                  solution.format_solution(self.settings.verbosity)
                );
```

Is the nested `format!` necessary here?